### PR TITLE
MM-24278 Fixes emoji picker keyboard navigation and highlight style

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Fixes emoji picker keyboard navigation and highlight style.
- On down arrow key,
  - If the cursor is at the beginning of the string, the down arrow should move the cursor to the end of the string
  [Other cases for up arrow key]
  - If Shift + CTRL/CMD + Arrow Down is pressed at any time, select/highlight the string to the right of the cursor.
  - Otherwise, move the selection down in the emoji picker.

- On up arrow key, 
  - If cursor is on text box, set the cursor to the beginning of the string.
  [Other cases for down arrow key]
  -  If Shift + CTRL/CMD + Up is pressed at any time, select/highlight the string to the left of the cursor.
  - If the cursor is highlighting an emoji in the top row, the up arrow should move the cursor back into the text box to the end of the string.
  - Otherwise, up arrow moves the emoji selector up a row.
- Selected emoji should be highlighted

#### Ticket Link
JIRA ticket - https://mattermost.atlassian.net/browse/MM-24278
Related issue: https://github.com/mattermost/mattermost-server/issues/11950

#### Related E2E Tests
- emoji_picker_keyboard_usability_spec.js
- emoji_keyboard_entry_spec.js

### CODE REVIEW

**Comment by saturninoabril** [comment]
@adamjclarkson Adding you on PM review since the regression was introduced by - https://github.com/mattermost/mattermost-webapp/pull/4292.  Particularly, please check if up and down arrow key on emoji picker are correct.  Thanks.

File: sass/components/_emoticons.scss [hunk 1]
```diff
@@ -309,11 +309,11 @@
 
     &.selected,
     &:hover {
-        background-color: v(center-channel-color-15);
+        background-color: v(center-channel-color-16);
     }
 
     &:active {
-        background-color: v(center-channel-color-30);
+        background-color: v(center-channel-color-32);
```
Comment (saturninoabril) [inline sass/components/_emoticons.scss:316]: CSS variables were no longer available, and so just pick the closest possible values - https://github.com/mattermost/mattermost-webapp/blob/master/utils/utils.jsx#L894-L915.  Adding @asaadmahmood for review, thanks

Comment (asaadmahmood) [reply sass/components/_emoticons.scss:316]: Yes, that is correct.

**Comment by asaadmahmood** [comment]
@saturninoabril I've already added the highlight thing in one of my PRs, so you can remove that if you want.
https://github.com/mattermost/mattermost-webapp/pull/5338

**Comment by saturninoabril** [comment]
@asaadmahmood Thanks, will update once it's merged.

**Comment by saturninoabril** [comment]
Thanks for the reviews. I'll get back to this PR and will add tests. Just fixing current issue with Cypress daily tests.

**Comment by saturninoabril** [comment]
@hmhealey E2E tests added.

File: e2e/cypress/integration/messaging/emoji_picker_keyboard_usability_spec.js [hunk 2]
```diff
@@ -64,11 +64,69 @@ describe('MM-13064 - Emoji picker keyboard usability', () => {
         cy.get('#emojiPickerAliasesPreview').should('have.text', ':smile_cat:');
     });
 
-    xit('If the down arrow key is pressed at any time, move the selection down in the emoji picker');
-    xit('If the up arrow is pressed when the emoji selector is already on the top row of emoji, set the cursor to the beginning of the string. Otherwise, up arrow moves the emoji selector up a row');
-    xit('If CTRL/CMD + Arrow Left/Up is pressed at any time, set the cursor to the beginning of the string when the cursor is at the end of the string');
-    xit('If CTRL/CMD + Arrow Right/Down is pressed at any time, set the cursor to the end of the string when the cursor is at the beginning of the string');
-    xit('If Shift + Arrow Left/Right is pressed at any time, should select or deselect individual characters based on the cursor position and if characters have or have not been selected');
-    xit('If Shift + CTRL/CMD + Arrow Left/Up is pressed at any time, selects the string to the left of the cursor');
-    xit('If Shift + CTRL/CMD + Arrow Right/Down is pressed at any time, selects the string to the right of the cursor');
```
Comment (saturninoabril) [inline e2e/cypress/integration/messaging/emoji_picker_keyboard_usability_spec.js:73]: Only related tests were  added. Previously skipped tests will be added on separate PR.
